runtime/v2/shim_load.go

```diff
@@ -180,6 +176,23 @@ func (m *ShimManager) loadShims(ctx context.Context) error {
 	return nil
 }
 
+func loadShimTask(ctx context.Context, bundle *Bundle, onClose func()) (_ *shimTask, retErr error) {
+	shim, err := loadShim(ctx, bundle, onClose)
+	if err != nil {
+		return nil, err
+	}
+	// Check connectivity, TaskService is the only required service, so create a temp one to check connection.
+	s, err := newShimTask(shim)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := s.PID(ctx); err != nil {
```

Comment:
suggest to follow original code to use timeout for checking connectivity.
In there, `ctx` is background and it won't timeout. If one is bad, it will be stuck forever.

Reply:
done, added a timeout.